FEATURE | Add generic parsing to DocBlockTypeAnnotations

bestform opened the pull request:
fixes #11

Commit:
FEATURE | Add generic parsing to DocBlockTypeAnnotations

fixes #11

bestform commented:
I recognized that the the travis build fails with --prefer-lowest on composer update. Looking into this...

j6s commented on tests/Parser/Visitor/Example/DocBlock/ImportedGenericArgument.php:
```diff
@@ -0,0 +1,10 @@
+<?php
```
I know it doesn't really matter in this class but every PHP file should `declare(strict_types=1);` - just to be consistent

bestform replied:
I left it out because none of the existing test classes declared it. But you are right. For good measure I've added the declaration to all the other ones as well.

j6s commented on src/Parser/Visitor/DocBlockTypeAnnotations.php:
```diff
@@ -52,9 +53,23 @@ private function extractDocBlocks(array $docBlocks, Context $context): void
+                    if ($type instanceof Collection) {
+                        $typesToResolve[] = new Object_($type->getFqsen());
+                        $typesToResolve[] = $type->getValueType();
+                    }
```
(This does not concernt your PR)

Reading the code I am fairly certain that it does not handle union types - both, in comments and actual type annotations (Coming in PHP8: https://wiki.php.net/rfc/union_types_v2 )

bestform replied:
It in fact does not. This should be a different issue, though.

j6s replied:
#15

Commit:
BUGFIX | Bump dependency

phpdocumentor/reflection-docblock has the option to install
a version of phpdocumentor/type-resolver below version 1.0

But the code now supports generic definitions in doc blocks
which are only supported in type-resolver version 1.0.0 and higher.

This commit also removes phpstan from the dev requirements as
it doesn't seem to be needed and moreover introduces class
ambiguity as it defines the same classes and namespaces as
refelction-docblock. And it is deprecated anyways.

Commit:
FIX | add strict types to all test classes

Commit:
FIX | Re-add phpstan as it is used in CI

bestform commented:
Oh, and I learned that the removed dependency was in fact being used in CI. So I re-added it.

j6s commented:
Oh, I forgot about this PR, sorry for waiting this long to merge.
